### PR TITLE
setup: pin pydantic-yaml and pydantic versions (CRAFT-718)

### LINKED ISSUE #158: ImportError: cannot import name 'YamlModel' from 'pydantic_yaml'
I believe this is because `setup.py` should pin `pydantic-yaml==0.4.2`, as this is also done in `requirements.txt` and `requirements-dev.txt`. Otherwise we are getting pydantic-yaml 0.5.1, for which the way of importing has changed.

```
build run-test: commands[0] | charmcraft build
Traceback (most recent call last):
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/bin/charmcraft", line 5, in <module>
    from charmcraft.main import main
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/charmcraft/main.py", line 26, in <module>
    from charmcraft import config, __version__
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/charmcraft/config.py", line 81, in <module>
    from charmcraft.parts import validate_part
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/charmcraft/parts.py", line 26, in <module>
    from craft_parts import LifecycleManager, Step, plugins
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/__init__.py", line 25, in <module>
    from .lifecycle_manager import LifecycleManager  # noqa: F401
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/lifecycle_manager.py", line 27, in <module>
    from craft_parts import errors, executor, packages, plugins, sequencer
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/executor/__init__.py", line 19, in <module>
    from .executor import ExecutionContext, Executor  # noqa: F401
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/executor/executor.py", line 35, in <module>
    from .part_handler import PartHandler
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/executor/part_handler.py", line 34, in <module>
    from craft_parts.state_manager import MigrationState, StepState, states
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/state_manager/__init__.py", line 19, in <module>
    from .state_manager import StateManager  # noqa: F401
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/state_manager/state_manager.py", line 32, in <module>
    from .states import StepState, get_step_state_path, load_step_state
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/state_manager/states.py", line 29, in <module>
    from .build_state import BuildState
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/state_manager/build_state.py", line 23, in <module>
    from .step_state import StepState, validate_hex_string
  File "/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/craft_parts/state_manager/step_state.py", line 23, in <module>
    from pydantic_yaml import YamlModel  # type: ignore
ImportError: cannot import name 'YamlModel' from 'pydantic_yaml' (/home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/lib/python3.9/site-packages/pydantic_yaml/__init__.py)
ERROR: InvocationError for command /home/lourot/Documents/git/canonical/charm-nova-compute-nvidia-vgpu/.tox/build/bin/charmcraft build (exited with code 1)

```

### PR DESCRIPTION
**Pull request by cmatsuoka**
Pin dependencies with API changes to allow craft-parts to run correctly.
Code will be updated to the current API in a separate PR. Pylint will
be unpinned after warnings issued by newer versions are addressed.

Fixes #158.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -69,7 +69,8 @@ def is_ubuntu() -> bool:
     "pydocstyle",
     # Incompatible with current pylint-fixme-info==1.0.2
     # https://github.com/PyCQA/pylint/issues/5390
-    "pylint<2.12.0",
+    # "pylint<2.12.0",
```
Comment (sergiusens) [inline setup.py:72]: this change is not explained.

**Review by sergiusens — CHANGES_REQUESTED** [review]
can you please explain the reason for changing the pinning for pylint and why is done as a code comment?

File: setup.py (hunk 1 above)
Comment (cmatsuoka) [reply setup.py:72]: I use pylint as part of my pre-push workflow (in `make tests`) and the newest version fails with extra warnings. The entry should eventually be restored to the commented one after those warnings are addressed (and the check can be added to the linters target in CI).

Comment (sergiusens) [reply setup.py:72]: Personally I would prefer to remove the line instead of commenting and a separate commit that we can later run `git revert` on

Comment (cmatsuoka) [reply setup.py:72]: Will do that.

Comment (cmatsuoka) [reply setup.py]: We can re-evaluate the need for pylint < 2.12 after addressing the issues, maybe pylint-fixme-info has already been updated to work with later versions.

Comment (cmatsuoka) [reply setup.py]: Will redo that in a separate commit and force-push to tidy things up.

Comment (cmatsuoka) [reply setup.py]: Ok, I won't pin pylint since I'm the only user. Once the warnings are addressed we can put it back in the linters target.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**